nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -92,11 +97,40 @@ def get_value_for_secret(cls, secret, obj=None, **kwargs):
                 raise exceptions.SecretProviderError(
                     secret, cls, "HashiCorp Vault invalid role_id and/or secret_id"
                 ) from err
+        elif auth_method == "kubernetes":
+            try:
+                client = hvac.Client(url=vault_settings["url"], verify=ca_cert)
+                with open(k8s_token_path, "r", encoding="utf-8") as token_file:
+                    jwt = token_file.read()
+                client.auth.kubernetes.login(role=vault_settings["role_name"], jwt=jwt)
+            except KeyError as err:
+                raise exceptions.SecretProviderError(
+                    secret, cls, "HashiCorp Vault configuration is missing a role_name"
+                ) from err
+            except hvac.exceptions.InvalidRequest as err:
+                raise exceptions.SecretProviderError(secret, cls, "HashiCorp Vault invalid role_name") from err
```

Comment:
If at all reasonably possible I would appreciate new tests for `auth_method == "kubernetes"` and `verify=ca_cert` options. Thanks!

Reply:
I have added test cases for `validate_vault_settings` and for `auth_method == "kubernetes"` however I can't add a useful testcase for `verify`.  The requests_mock ignores https and treats it as http therefore all https requests will be returned as valid.

Reply:
@nniehoff Thanks! The requests-mock docs appear to support `verify=`, but I trust you. https://requests-mock.readthedocs.io/en/latest/release-notes.html?highlight=verify#relnotes-1-5-1-new-features